memory_test.go

```diff
@@ -395,6 +395,15 @@ func TestMemCheck(t *testing.T) {
 			expectedSizeDiff: testNativeValueMemUsage +
 				(2 + SizeString), // "nv",
 		},
+		{
+			desc: "array.map function",
+			script: `x = [1,2,3,4,5]
+			x.map(_ => {
+				checkMem();
+				return "a"
+			})`,
+			expectedSizeDiff: 4 * (1 + SizeString),
```

Comment:
are we multiplying by 4 because we're checking the memory before returning the last element (number 5)?

Reply:
Yeah that's correct